### ISSUE TITLE
Allow optimizer to function when external data is not available

### ISSUE THREAD

**Issue by justinchuby** [microsoft/onnxscript#1792]
Models with external data may not have their data available when being passed into the optimizer. We should still optimize what we can without raising an error.

**Comment by justinchuby** [comment]
cc @gramalingam

**Comment by gramalingam** [comment]
I took a quick look at the IR-based constant-folder/optimizer. I think it should work, as long as the `numpy()` method returns None or value.const_value is None. We can change it to use whatever appropriate methods the IR exposes to access such external tensors, allowing for possibility that the value may not be available.

On a different note: I just remembered that ir.Value.const_value is used for initializers that are graph inputs as well. This could break a few things. The optimizer etc. assume `.const_value` has a value only for constants, but graph inputs that are initializers should NOT be treated as constants.

**Comment by gramalingam** [comment]
I do see an error message `Data of TensorProto ( tensor name: model/model/decoder/Constant_1_attr::value) should be stored in model_model_decoder_Constant_1_attr__value but it doesn't exist or is not accessible.` for ir-based optimization of BlenderbotForConditionalGeneration.

**Comment by justinchuby** [comment]
> I took a quick look at the IR-based constant-folder/optimizer. I think it should work, as long as the `numpy()` method returns None or value.const_value is None. We can change it to use whatever appropriate methods the IR exposes to access such external tensors, allowing for possibility that the value may not be available.
> 
> On a different note: I just remembered that ir.Value.const_value is used for initializers that are graph inputs as well. This could break a few things. The optimizer etc. assume `.const_value` has a value only for constants, but graph inputs that are initializers should NOT be treated as constants.

Right. We shouldn’t assume only Constants have const_value. Any value can have constant values.
